### PR TITLE
SNOW-683057: Improve code coverage of row, files, session, functions

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-683057

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/files.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def close(self) -> None:
         """
         return
 
-    def detach(self) -> None:
+    def detach(self) -> None:  # pr
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/files.py:65]: nit: incomplete comment?

File: src/snowflake/snowpark/row.py [hunk 2]
```diff
@@ -175,10 +173,13 @@ def __repr__(self):
             return "Row({})".format(
                 ", ".join(f"{k}={v!r}" for k, v in zip(self._fields, self))
             )
-        elif self._named_values:
+        elif self._named_values:  # pragma: no cover
+            # this might not be reachable because when there is self._named_values, there is always self._fields
+            # Need to review later.
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/row.py:178]: Create a ticket for this?

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/files.py (hunk 1 above)
Comment (sfc-gh-yixie) [reply src/snowflake/snowpark/files.py]: good catch. I removed it. This method is actually covered.

File: src/snowflake/snowpark/row.py (hunk 2 above)
Comment (sfc-gh-yixie) [reply src/snowflake/snowpark/row.py:178]: Created

**Pull request merged**